app/models/concerns/hyrax/collection_behavior.rb

```diff
@@ -78,8 +89,39 @@ def member_object_ids
       ActiveFedora::Base.search_with_conditions("member_of_collection_ids_ssim:#{id}").map(&:id)
     end
 
+    protected
+
+      def collection_type=(collection_type)
+        @collection_type = collection_type
```

Comment:
I used ActiveRecord::AssociationTypeMismatch because it seems close in concept.  But there isn't a real associate between the two other than a runtime association.